crates/cache/src/lib.rs

```diff
@@ -77,24 +79,34 @@ pub trait Sizeable {
     fn get_memory_size(&self) -> usize;
 }
 
+impl Sizeable for Vec<Vec<f32>> {
+    fn get_memory_size(&self) -> usize {
+        self.iter()
+            .map(|vec| vec.len() * std::mem::size_of::<f32>())
+            .sum()
+    }
+}
+
 pub trait HashProvider {
     fn hasher(&self) -> Box<dyn Hasher>;
 }
 
 /// Trait for types that can be converted to a set of table references.
 pub trait AsTableRefs {
```

Comment:
Should we just not implement this for the new embedding cache? Seems weird to make this change.